### PR TITLE
handled score extension images

### LINKED ISSUE #1418: Score Extension images are not handled properly
Notes are handled properly at:
https://af.wikiquote.org/api/rest_v1/page/html/Johann_Sebastian_Bach

but not at:
http://library.kiwix.org/wikiquote_af_all_maxi/A/Johann_Sebastian_Bach

### PR DESCRIPTION
**Pull request by MananJethwani**
@kelson42 this resolves #1418

### CODE REVIEW

**Comment by MananJethwani** [comment]
@kelson42 sorry I forgot about test please don't review yet

**Comment by MananJethwani** [comment]
@kelson42 done.....I am using only `.png` in regex as it is mentioned [here](https://www.mediawiki.org/wiki/Extension:Score#:~:text=The%20Score%20extension%20allows%20the,into%20audio%20and%20MIDI%20files.) that this extension generates only png images

**Comment by kelson42** [comment]
@MananJethwani Code factor does not pass

**Comment by MananJethwani** [comment]
@kelson42 please have a look now

File: src/util/misc.ts [hunk 1]
```diff
@@ -233,8 +233,15 @@ export function getMediaBase(url: string, escape: boolean) {
   let parts;
   let filename;
 
+  // Score images(music notes)
+  parts = SCORE_IMAGE_URL_REGEX.exec(decodedUrl)
```
Comment (kelson42) [inline src/util/misc.ts:237]: Missing semi colon

File: src/util/misc.ts [hunk 2]
```diff
@@ -233,8 +233,15 @@ export function getMediaBase(url: string, escape: boolean) {
...
+  if(parts !== null){
```
Comment (kelson42) [inline src/util/misc.ts:238]: Missing space

**Review by kelson42 — CHANGES_REQUESTED** [review]
The problem of this PR is that score images are 0,0001% of all images, but mwoffliner checks them first, big waste of CPU.

**Comment by MananJethwani** [comment]
> The problem of this PR is that score images are 0,0001% of all images, but mwoffliner checks them first, big waste of CPU.

I would try to somehow merge the first 2 conditions to make this a bit faster I might have an idea....give me some time

**Comment by kelson42** [comment]
You should make the thumb regex more specific and put your score regex after

**Comment by MananJethwani** [comment]
@kelson42 instead of moving it below thumb regex I could do something like this also
```
const len = 'https://upload.wikimedia.org/'.length;
str.indexOf('score/', len) === len;
```
or change the pattern only using
```
const pattern = str.includes('/score/') ? ... : ...;
```
if these are slow according to you I will go with your solution only of moving score regex down as conditional regex don't work in js

**Comment by kelson42** [comment]
No, do a proper regex please and don't test only the keyword "score".

**Review by kelson42 — APPROVED** [review]

**Pull request merged**